pkg/safeguards/helpers.go

```diff
@@ -0,0 +1,268 @@
+		if err != nil {
+			return nil, fmt.Errorf("could not add constraint: %w", err)
+		}
+
+		constraints = append(constraints, u)
+	}
+
+	return constraints, nil
+}
+
+func (fc FileCrawler) ReadConstraint(name string) (*unstructured.Unstructured, error) {
+	var constraint *unstructured.Unstructured
+
+	for _, sg := range fc.Safeguards {
+		if sg.name == name {
+			c, err := reader.ReadConstraint(fc.constraintFS, sg.constraintPath)
+			if err != nil {
+				return nil, fmt.Errorf("could not add constraint: %w", err)
+			}
+
+			constraint = c
+		}
+	}
+	if constraint == nil {
+		return nil, fmt.Errorf("no constraint exists with name: %s", name)
+	}
+
+	return constraint, nil
+}
+
+// loads constraint templates, constraints into constraint client
+func loadConstraintTemplates(ctx context.Context, c *constraintclient.Client, constraintTemplates []*templates.ConstraintTemplate) error {
+	// AddTemplate adds the template source code to OPA and registers the CRD with the client for
+	// schema validation on calls to AddConstraint. On error, the responses return value
+	// will still be populated so that partial results can be analyzed.
+	for _, ct := range constraintTemplates {
+		_, err := c.AddTemplate(ctx, ct)
+		if err != nil {
+			return fmt.Errorf("could not add template: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func loadConstraints(ctx context.Context, c *constraintclient.Client, constraints []*unstructured.Unstructured) error {
+	// AddConstraint validates the constraint and, if valid, inserts it into OPA.
+	// On error, the responses return value will still be populated so that
+	// partial results can be analyzed.
+	for _, con := range constraints {
+		_, err := c.AddConstraint(ctx, con)
+		if err != nil {
+			return fmt.Errorf("could not add constraint: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func loadManifestObjects(ctx context.Context, c *constraintclient.Client, objects []*unstructured.Unstructured) error {
+	// AddData inserts the provided data into OPA for every target that can handle the data.
+	// On error, the responses return value will still be populated so that
+	// partial results can be analyzed.
+	for _, o := range objects {
+		_, err := c.AddData(ctx, o)
+		if err != nil {
+			return fmt.Errorf("could not add data: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// IsDirectory determines if a file represented by path is a directory or not
+func IsDirectory(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+
+	return fileInfo.IsDir(), nil
+}
+
+// IsYAML determines if a file is of the YAML extension or not
+func IsYAML(path string) bool {
+	return filepath.Ext(path) == ".yaml" || filepath.Ext(path) == ".yml"
+}
+
+// GetManifestFiles uses filepath.Walk to retrieve a list of the manifest files within the given manifest path
+func GetManifestFiles(p string) ([]ManifestFile, error) {
+	var manifestFiles []ManifestFile
+
+	noYamlFiles := true
```

Comment:
couldn't this be accomplished just by checking len(manifestFiles)?

Reply:
yes